homeassistant/components/unifi/device_tracker.py

```diff
@@ -313,46 +389,119 @@ async def options_updated(self) -> None:
             await self.remove_item({self.client.mac})
 
 
-class UniFiDeviceTracker(UniFiBase, ScannerEntity):
-    """Representation of a network infrastructure device."""
+class UnifiScannerEntity(ScannerEntity, Generic[_HandlerT, _DataT]):
+    """Representation of a UniFi scanner."""
 
-    DOMAIN = DOMAIN
-    TYPE = DEVICE_TRACKER
+    entity_description: UnifiEntityDescription[_HandlerT, _DataT]
+    _attr_should_poll = False
 
-    def __init__(self, device, controller):
-        """Set up tracked device."""
-        super().__init__(device, controller)
+    def __init__(
+        self,
+        obj_id: str,
+        controller: UniFiController,
+        description: UnifiEntityDescription[_HandlerT, _DataT],
+    ) -> None:
+        """Set up UniFi scanner entity."""
+        self._obj_id = obj_id
+        self.controller = controller
+        self.entity_description = description
 
-        self.device = self._item
-        self._is_connected = device.state == 1
         self._controller_connection_state_changed = False
+        self._removed = False
         self.schedule_update = False
 
+        self._attr_available = description.available_fn(controller, obj_id)
+        self._attr_unique_id: str = description.unique_id_fn(obj_id)
+
+        obj = description.object_fn(controller.api, obj_id)
+        self._is_connected = description.is_connected_fn(controller.api, obj)
+        self._attr_name = description.name_fn(obj)
+
+    @property
+    def is_connected(self):
+        """Return true if the device is connected to the network."""
+        return self._is_connected
+
+    @property
+    def hostname(self) -> str | None:
+        """Return hostname of the device."""
+        return self.entity_description.hostname_fn(self.controller.api, self._obj_id)
+
+    @property
+    def ip_address(self) -> str:
+        """Return the primary ip address of the device."""
+        return self.entity_description.ip_address_fn(self.controller.api, self._obj_id)
+
+    @property
+    def mac_address(self) -> str:
+        """Return the mac address of the device."""
+        return self._obj_id
+
+    @property
+    def source_type(self) -> SourceType:
+        """Return the source type, eg gps or router, of the device."""
+        return SourceType.ROUTER
+
+    @property
+    def unique_id(self) -> str:
+        """Return a unique ID."""
+        return self._attr_unique_id
+
```

Comment:
This is default for an Entity, so not needed?

Reply:
Device tracker is special :)